modules/discretedata/connectivity/structuredgrid.cpp

```diff
@@ -0,0 +1,264 @@
+/*********************************************************************************
+*
+* Inviwo - Interactive Visualization Workshop
+*
+* Copyright (c) 2012-2018 Inviwo Foundation
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+* 2. Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following disclaimer in the documentation
+* and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+* ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*********************************************************************************/
+
+#include "structuredgrid.h"
+#include "discretedata/channels/analyticchannel.h"
+#include "discretedata/connectivity/elementiterator.h"
+
+namespace inviwo {
+namespace dd {
+
+StructuredGrid::StructuredGrid(GridPrimitive gridDimension, const std::vector<ind>& numCellsPerDim)
+    : Connectivity(gridDimension), numCellsPerDimension_(numCellsPerDim) {
+    ivwAssert(numCellsPerDim.size() == gridDimension,
+              "Grid dimension should match cell dimension.");
+    numCellsPerDimension_ = std::vector<ind>(numCellsPerDim);
+
+    ind numCells = 1;
+    ind numVerts = 1;
+    for (ind dim = GridPrimitive::Vertex; dim < gridDimension; ++dim) {
+        numCells *= numCellsPerDimension_[dim];
+        numVerts *= numCellsPerDimension_[dim] + 1;
+    }
+
+    numGridPrimitives_[gridDimension] = numCells;
+    numGridPrimitives_[GridPrimitive::Vertex] = numVerts;
+}
```

Comment:
Don't use need to initialize the Connectivity::vertices_ member? or where is that done?